FlowCal/plot.py

```diff
@@ -378,31 +381,44 @@ def density2d(data,
         raise ValueError('two channels need to be specified')
     data_plot = data[:, channels]
 
-    # If bins are not specified, try to get bins from data object
-    if (bins is None and hasattr(data_plot, 'hist_bin_edges')
-            and data_plot.hist_bin_edges(0) is not None
-            and data_plot.hist_bin_edges(1) is not None):
-        # Get bin information
-        bdx = data_plot.hist_bin_edges(0)
-        bdy = data_plot.hist_bin_edges(1)
-        # Get bin scaled indices
-        xdx = np.linspace(0, 1, len(bdx))
-        xsx = np.linspace(0, 1, (len(bdx) - 1)/div + 1)
-        xdy = np.linspace(0, 1, len(bdy))
-        xsy = np.linspace(0, 1, (len(bdy) - 1)/div + 1)
-        # Generate sub-sampled bins
-        bins = np.array([np.interp(xsx, xdx, bdx), 
-                            np.interp(xsy, xdy, bdy)])
+    # If ``data_plot.hist_bins()`` exists, obtain bin edges from it if
+    # necessary.
+    if hasattr(data_plot, 'hist_bins'):
```

Comment:
same as above. not sure if you want to update this test to acknowledge that `hist_bins` should be callable.